pyaedt/icepak.py

```diff
@@ -3779,6 +3778,108 @@ def assign_source(
             self.boundaries.append(bound)
             return bound
 
+    @pyaedt_function_handler()
+    def create_network_object(self, name=None, props=None, create=False):
+        """Create a thermal network.
+
+        Parameters
+        ----------
+        name : str, optional
+            String with the name of the network object. Default is ``None``.
+        props : dict, optional
+            Dictionary with information required by oModule.AssignNetworkBoundary. Default is ``None``.
+        create : bool, optional
+            Whether to create immediately the network inside AEDT or not. Default is ``False`` and the user
+            can modify the network from pyAEDT functions and create the network only afterwards.
+
+        Returns
+        -------
+        :class:`pyaedt.modules.Boundary.BoundaryNetwork`
+            Boundary network object when successful or ``None`` when failed.
+
+        References
+        ----------
+
+        >>> oModule.AssignNetworkBoundary
+
+        Examples
+        --------
+
+        >>> from pyaedt import Icepak
+        >>> app = Icepak()
+        >>> network = app.create_network_object()
+        """
+        bound = NetworkObject(self, name, props, create)
+        self.boundaries.append(bound)
+        return bound
+
+    @pyaedt_function_handler()
+    def create_resistor_network_from_matrix(self, sources_power, faces_ids, matrix, network_name=None, node_names=None):
+        """Create a thermal network.
+
+        Parameters
+        ----------
+        sources_power : list of str or list of float
+            List containing all the power value of the internal nodes. If the element of the list
+            is a float, "W" unit is assigned.
+        faces_ids :  list of int
+            List containing all the face ids that are network nodes.
+        matrix : list of list
+            Strict lower square triangular matrix containing the links value between the nodes of the network.
+            If the element of the matrix is a float "cel_per_w" unit is automatically assigned,
+            else the unit prescribed in the string is used. The element of the matrix in the i-th row
+            and j-th column is the link value between the i-th node and j-th node. The list of nodes is
+            automatically created from sources_power and faces_ids list (in this order).
+        network_name : str, optional
+            Name of the network boundary. The default is ``None`` and the boundary name will
+            be generated automatically.
+        node_names : list of str, optional
+            List of the name of all the nodes in the network.
```

Comment:
```suggestion
        sources_power : list of str or list of float
            List containing all the power value of the internal nodes. If the element of the list
            is a float, "W" unit is assigned.
        faces_ids :  list of int
            List containing all the face IDs that are network nodes.
        matrix : list of list
            Strict lower-square triangular matrix containing the link values between
            the nodes of the network. If the element of the matrix is a float, the
            ``cel_per_w`` unit is automatically assigned. Otherwise, the unit specified
            in the string is used. The element of the matrix in the i-th row
            and j-th column is the link value between the i-th node and j-th node.
            The list of nodes is automatically created from the lists for the
            ``sources_power`` and ``faces_ids`` parameters (in this order).
        network_name : str, optional
            Name of the network boundary. The default is ``None``, in which
            case the boundary name is generated automatically.
        node_names : list of str, optional
            Name of all the nodes in the network.
```